lib/puppet/network/http/connection.rb

```diff
@@ -56,7 +56,7 @@ def initialize(host, port, options = {})
 
       options = OPTION_DEFAULTS.merge(options)
       @use_ssl = options[:use_ssl]
-      @verify = options[:verify]
+      @verifier = Puppet::SSL::VerifierAdapter.new(options[:verify])
```

Comment:
what happens if `options[:verify]` is already the new `verifier` rather than the old `verify`?

Reply:
that's not great wording.

what if `options[:verify]` is already a `Verifier` rather than the old `Puppet::SSL::Validator`?

Reply:
I think it's unlikely, because callers don't call `Connection.new` directly. Instead they create a connection from the `HttpPool` factory whose methods supply the correct verifier or validator.

Reply:
Given the confusion around `verify` vs `verifier` vs `validator` I could imagine someone passing in a `Validator` when a `Verifier` is expected or vice-versa. Part of the problem is the [`DefaultValidator` doesn't extend `Validator`](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/ssl/validator/default_validator.rb#L9) so I can't just check for `is_a?(Puppet::SSL::Validator)`. Also the `http_ssl_instance` method allows arbitrary validators to be passed in. But I could check for the case where a `Verifier` is passed in, and make sure that's used correctly.

Reply:
Updated